src/App.tsx

```diff
@@ -1,39 +1,54 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
-
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component {
+  state = {
```

Comment:
Generally it's a good practice to type your state
```tsx
interface AppState {
  hasClock: boolean;
  clockName: string;
  clockNameInterval: number;
}

export class App extends React.Component<{}, AppState> {
  state: Readonly<AppState> = {
    hasClock: true,
    clockName: 'Clock-0',
    clockNameInterval: -1,
  }
```
ReadOnly generic type will signify the next developer that he is working with state and that he should not modify it explicitly without using setState, `this.state.hasClock = false` will then correctly throw a typescript error

Reply:
I added readonly type.